mops-torch/src/hpe.cpp

```diff
@@ -0,0 +1,90 @@
+#include "mops/torch/hpe.hpp"
+#include "mops/torch/utils.hpp"
+
+using namespace mops_torch;
+
+
+torch::Tensor mops_torch::homogeneous_polynomial_evaluation(
+    torch::Tensor A,
+    torch::Tensor C,
+    torch::Tensor indices_A
+) {
+    return HomogeneousPolynomialEvaluation::apply(A, C, indices_A);
+}
+
+torch::Tensor HomogeneousPolynomialEvaluation::forward(
+    torch::autograd::AutogradContext *ctx,
+    torch::Tensor A,
+    torch::Tensor C,
+    torch::Tensor indices_A
+) {
+    // TODO: check tensor shapes
+
```

Comment:
we should error if `C` requires gradient here, or implement the corresponding gradients

Reply:
You're right. We'll throw an error. `C` is supposed to be constant in the three operations that use it. I would not implement its gradients, at least for now